src/cljs/owlet_ui/components/branch.cljs

```diff
@@ -4,6 +4,20 @@
             [reagent.core :as reagent :refer [atom]]
             [camel-snake-kebab.core :refer [->kebab-case]]))
 
+(def hover-image (reagent/atom ""))
+
+(defn get-random [images]
+  (get images (rand-int (count images))))
+
+(defn set-hover [images]
+  (let [image (get-random images)]
+    (if (> (count images) 1)
+      ; prevent the same image twice in a row
+      (if (== image @hover-image)
+        (set-hover images)
```

Comment:
Not to be a code cop (OK, I am!) but ... in Clojure/ClojureScript, It's idiomatic and more efficient to use recur in a situation like this. The result from your recursive call is simply returned from set-hover, i.e., set-hover does no further calculation with the result. In this case, you can effectively employ tail-call optimization by calling `(recur images)` instead of `(set-hover images)`. Besides not accumulating stack and being faster, the explicitness of using recur makes your code more readable, and it invites you think about how it really works! [Docs here.](http://cljs.github.io/api/cljs.core/recur)